cmd/argocd/commands/cert.go

```diff
@@ -0,0 +1,305 @@
+				errors.CheckError(fmt.Errorf("No valid SSH known hosts data found."))
+			}
+
+			for _, knownHostsEntry := range sshKnownHostsLists {
+				hostname, certSubType, certData, err := certutil.TokenizeSSHKnownHostsEntry(knownHostsEntry)
+				errors.CheckError(err)
+				_, _, err = certutil.KnownHostsLineToPublicKey(knownHostsEntry)
+				errors.CheckError(err)
+				certificate := appsv1.RepositoryCertificate{
+					ServerName:  hostname,
+					CertType:    "ssh",
+					CertSubType: certSubType,
+					CertData:    certData,
+				}
+
+				certificates = append(certificates, certificate)
+			}
+
+			certList := &appsv1.RepositoryCertificateList{Items: certificates}
+			response, err := certIf.Create(context.Background(), &certificatepkg.RepositoryCertificateCreateRequest{
+				Certificates: certList,
+				Upsert:       upsert,
+			})
+			errors.CheckError(err)
+			fmt.Printf("Successfully created %d SSH known host entries\n", len(response.Items))
+		},
+	}
+	command.Flags().StringVar(&fromFile, "from", "", "Read SSH known hosts data from file (default is to read from stdin)")
+	command.Flags().BoolVar(&batchProcess, "batch", false, "Perform batch processing by reading in SSH known hosts data (mandatory flag)")
+	command.Flags().BoolVar(&upsert, "upsert", false, "Replace existing SSH server public host keys if key is different in input")
+	return command
+}
+
+// NewCertRemoveCommand returns a new instance of an `argocd cert rm` command
+func NewCertRemoveCommand(clientOpts *argocdclient.ClientOptions) *cobra.Command {
+	var (
+		removeAllCerts bool
+		certType       string
+		certSubType    string
+		certQuery      certificatepkg.RepositoryCertificateQuery
+	)
+	var command = &cobra.Command{
+		Use:   "rm REPOSERVER",
+		Short: "Remove certificate of TYPE for REPOSERVER",
+		Run: func(c *cobra.Command, args []string) {
+			if len(args) < 1 && !removeAllCerts {
+				c.HelpFunc()(c, args)
+				os.Exit(1)
+			}
+			conn, certIf := argocdclient.NewClientOrDie(clientOpts).NewCertClientOrDie()
+			defer util.Close(conn)
+			if removeAllCerts {
+				certQuery = certificatepkg.RepositoryCertificateQuery{
+					HostNamePattern: "*",
+					CertType:        "*",
+					CertSubType:     "*",
+				}
+			} else {
+				certQuery = certificatepkg.RepositoryCertificateQuery{
+					HostNamePattern: args[0],
+					CertType:        certType,
+					CertSubType:     certSubType,
+				}
+			}
+			removed, err := certIf.Delete(context.Background(), &certQuery)
+			errors.CheckError(err)
+			if len(removed.Items) > 0 {
+				for _, cert := range removed.Items {
+					fmt.Printf("Removed cert for '%s' of type '%s' (subtype '%s')\n", cert.ServerName, cert.CertType, cert.CertSubType)
+				}
+			} else {
+				fmt.Println("No certificates were removed (none matched the given patterns)")
+			}
+		},
+	}
+	command.Flags().BoolVar(&removeAllCerts, "remove-all", false, "Remove all configured certificates of all types from server (DANGER: use with care!)")
```

Comment:
It was convenient during test cycles, but you are right, it's quite a dangerous option (therefore the fat DANGER sign). However, the same could be accomplished by using a `*` wildcard as the repo server name, which is not even really documented. 

I see two options:

1. Ask the user with a prompt if he really wants to perform that dangerous operation when `--remove-all` is specified or `*` was given as the repo-server name without any other selector such as `--cert-type`
2. Discard both possibilities for good, and have the user delete data from ConfigMap objects if they really want to get rid of all certificates.

Thinking of it, sh*t happens too often and I guess variant 2 is the better one. Will remove the code from the CLI. Do you think we should check on server side as well, whether '*' was specified in case someone writes against the API?